packages/selenium/test/RunPartial/FinishRunTests.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Linq;
+using System.IO;
+using Deque.AxeCore.Commons;
+using Deque.AxeCore.Selenium;
+using NUnit.Framework;
+using Newtonsoft.Json;
+
+namespace Deque.AxeCore.Selenium.Test.RunPartial
+{
+    public class FinishRunTests : TestBase
+    {
+        protected string axeForceLegacy { get; set; } = null;
+
+        [OneTimeSetUp]
+        public void SetUpForceLegacy()
+        {
+            axeForceLegacy = File.ReadAllText(
+                Path.Combine(TestContext.CurrentContext.TestDirectory, "fixtures/axe-force-legacy.js")
+            );
+        }
+
+        [Test]
+        public void ShouldIsolateCallToFinishRun()
+        {
+            GoToFixture("isolated-finish.html");
+            var axe = new AxeBuilder(driver);
+            Assert.DoesNotThrow(() => axe.Analyze());
+        }
+
+        [Test]
+        public void ShouldHaveSameResultsAsLegacy()
+        {
+            GoToFixture("nested-iframes.html");
+
+            var legacyResults = new AxeBuilder(driver, CustomSource($"{axeSource}{axeForceLegacy}"))
+                .Analyze();
+
+            Assert.That(legacyResults.TestEngineName, Is.EqualTo("axe-legacy"));
+
+
+            GoToFixture("nested-iframes.html");
+            var runPartialResults = new AxeBuilder(driver)
+                .Analyze();
+
+            Assert.That(legacyResults.Violations.Length, Is.EqualTo(runPartialResults.Violations.Length));
+            for (int i = 0; i < legacyResults.Violations.Length; i++)
+            {
+                AssertAxeItemsEqual(legacyResults.Violations[i], runPartialResults.Violations[i], true);
+            }
+
+            Assert.That(legacyResults.Passes.Length, Is.EqualTo(runPartialResults.Passes.Length + 1));
```

Comment:
It's not clear from reading this why it's expected for the behavior of the bypass rule to different between the two engines - it would be helpful to leave a comment referencing why the difference in behavior would be expected